src/App.tsx

```diff
@@ -1,14 +1,68 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState('all');
+  const [searchBarValue, setSearchBarValue] = useState('');
+
+  useEffect(() => {
+    getTodos().then(data => setTodos(data));
+  }, []);
+
+  const showTodo = useCallback((todo: Todo) => {
+    setSelectedTodo(todo);
+  }, []);
```

Comment:
do we really need to cover this function in useCallback. I think we can remove it

Reply:
Well, in our case, Todolist component is rerendred all time. But hypotheticaly, we can have menu component for example, which will rerender app, but will not effect todolist. That is why i wrapped this handler, so Todolist won`t be rerendred just because its parent component is rerendered.